insights/client/utilities.py

```diff
@@ -471,7 +476,45 @@ def os_release_info():
             continue
         except Exception as e:
             logger.warning("Failed to detect OS version: %s", e)
-    return (os_family, os_release)
+    return os_family, os_release
+
+
+def get_rhel_version():
+    # type: () -> int
+    """Reads /etc/os-release to determine RHEL version.
+
+    For RHEL, the major version is always returned.
+
+    For CentOS, the major version is always returned.
+
+    For Fedora, the latest known RHEL version is returned.
+    RHEL 10 was branched from Fedora 40; until RHEL 11 is branched,
+    this method returns `10` for all Fedora versions.
+
+    :returns: Major RHEL version (or an equivalent for non-RHEL systems)
+    :raises ValueError: Version could not be determined.
+    """
+    distribution, release = os_release_info()
+    if distribution == "Unknown":
+        raise ValueError("Could not determine distribution family.")
+    if release == "":
+        raise ValueError("Could not determine version of {}.".format(distribution))
+
+    _distribution = distribution.lower()  # type: str
+    version = 0  # type: int
+    for family_name in ("red hat enterprise linux", "centos"):
+        if family_name in _distribution:
+            version = int(release.split(".")[0])
+    for family_name in ("fedora", ):
+        if family_name in _distribution:
+            version = 10
```

Comment:
What about using `.startswith()` instead of for loops? Something like:
```py
if _distribution.startswith("red hat enterprise linux") or _distribution.startswith("centos"):
    ...
elif _distribution.startswith("fedora"):
    ....
else:
    # raise a ValueError
```

Reply:
I wanted to keep it scalable, but perhaps it's just overengineering. Will update.